Update aws-sam-deployments.md (#6435)
harness/developer-hub: docs/continuous-delivery/deploy-srv-diff-platforms/aws/aws-sam-deployments.md

Harness automatically generates a containerized step group containing the steps needed for the SAM build and deploy.

> When the step group setting **Enable container based execution** is enabled, the step group is containerized. SAM runs on containerized step groups only. You need not install SAM CLI on Harness Delegate as SAM steps are baked in with the SAM CLI. Hence, the SAM steps run in the containers that re brought up inside the pod within the Kubernetes cluster configured in the step group configuration. 

Reviewer: " as SAM steps are baked" -> " as Harness provided SAM step images are baked"

Reviewer: "Hence, the SAM steps run" -> "SAM steps run"

> When the step group setting **Enable container based execution** is enabled, the step group is containerized. SAM runs on containerized step groups only. You need not install SAM CLI on Harness Delegate as SAM steps are baked in with the SAM CLI. Hence, the SAM steps run in the containers that re brought up inside the pod within the Kubernetes cluster configured in the step group configuration. 

Author: ```suggestion
When the step group setting **Enable container based execution** is enabled, the step group is containerized. SAM runs on containerized step groups only. You need not install SAM CLI on Harness Delegate as Harness provided SAM step images are baked within the SAM CLI. SAM steps run in the containers that are brought up inside the pod within the Kubernetes cluster configured in the step group configuration. 
```

> When the **Enable container based execution** setting is disabled, Harness Delegate performs all tasks on its host system.

Reviewer 2: I think the point here is to mention that, if **Enable container based execution** setting is disabled, there is no sense in using the SAM build/deploy steps within the step group. The delegate can't do SAM deployment. 
cc: @piyush-bhuwalka

> When the **Enable container based execution** setting is disabled, Harness Delegate performs all tasks on its host system.

Author: ```suggestion
When the **Enable container based execution** setting is disabled, Harness Delegate cannot perform the SAM deployment.
```

> When the **Enable container based execution** setting is disabled, the Harness Delegate performs all tasks on its host system.

Author: ```suggestion
When the **Enable container based execution** setting is disabled, Harness Delegate performs all tasks on its host system.
```
